lib/ansible/modules/network/cloudengine/ce_vlan.py

```diff
@@ -0,0 +1,692 @@
+
+        conf_str = CE_NC_GET_VLANS
+        xml_str = get_nc_config(self.module, conf_str)
+        vlan_list = list()
+
+        if "<data/>" in xml_str:
+            return vlan_list
+        else:
+            vlan_list = re.findall(
+                r'.*<vlanId>(.*)</vlanId>.*', xml_str)
+            return vlan_list
+
+    def vlan_series(self, vlanid_s):
+        """ convert vlan range to list """
+
+        vlan_list = []
+        peerlistlen = len(vlanid_s)
+        if peerlistlen != 2:
+            self.module.fail_json(msg='Error: Format of vlanid is invalid.')
+        for num in range(peerlistlen):
+            if not vlanid_s[num].isdigit():
+                self.module.fail_json(
+                    msg='Error: Format of vlanid is invalid.')
+        if int(vlanid_s[0]) > int(vlanid_s[1]):
+            self.module.fail_json(msg='Error: Format of vlanid is invalid.')
+        elif int(vlanid_s[0]) == int(vlanid_s[1]):
+            vlan_list.append(str(vlanid_s[0]))
+            return vlan_list
+        for num in range(int(vlanid_s[0]), int(vlanid_s[1])):
+            vlan_list.append(str(num))
+        vlan_list.append(vlanid_s[1])
+
+        return vlan_list
+
+    def vlan_region(self, vlanid_list):
+        """ convert vlan range to vlan list """
+
+        vlan_list = []
+        peerlistlen = len(vlanid_list)
+        for num in range(peerlistlen):
+            if vlanid_list[num].isdigit():
+                vlan_list.append(vlanid_list[num])
+            else:
+                vlan_s = self.vlan_series(vlanid_list[num].split('-'))
+                vlan_list.extend(vlan_s)
+
+        return vlan_list
+
+    def vlan_range_to_list(self, vlan_range):
+        """ convert vlan range to vlan list """
+
+        vlan_list = self.vlan_region(vlan_range.split(','))
+
+        return vlan_list
+
+    def vlan_list_to_bitmap(self, vlanlist):
+        """ convert vlan list to vlan bitmap """
+
+        vlan_bit = ['0'] * 1024
+        bit_int = [0] * 1024
+
+        vlan_list_len = len(vlanlist)
+        for num in range(vlan_list_len):
+            tagged_vlans = int(vlanlist[num])
+            if tagged_vlans <= 0 or tagged_vlans > 4094:
+                self.module.fail_json(
+                    msg='Error: Vlan id is not in the range from 1 to 4094.')
+            j = tagged_vlans / 4
+            bit_int[j] |= 0x8 >> (tagged_vlans % 4)
+            vlan_bit[j] = hex(bit_int[j])[2]
+
+        vlan_xml = ''.join(vlan_bit)
+
+        return vlan_xml
+
+    def check_params(self):
+        """Check all input params"""
+
+        # is params invalid
+        if not self.vlan_id and not self.vlan_range:
+            self.module.fail_json(
+                msg='Error: Vlan_id or vlan_range must be set.')
+        elif self.vlan_id and self.vlan_range:
+            self.module.fail_json(
+                msg='Error: Vlan_id or vlan_range can not be set at the same time.')
+
+        if not self.vlan_id and self.description:
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
mutually_exclusive
required_together
required_one_of
require_if
```